stacks/udi/devfile.yaml

```diff
@@ -0,0 +1,29 @@
+schemaVersion: 2.2.0
+metadata:
+  name: udi
+  displayName: Universal Developer Image
+  description: Universal Developer Image provides various programming languages tools and runtimes for instant coding
+  icon: https://landscape.cncf.io/logos/devfile.svg
+  tags:
+    - Java
+    - Maven
+    - Scala
+    - PHP
+    - .NET
+    - Node.js
+    - Go
+    - Python
+    - Pip
+    - ubi8
+  projectType: default
```

Comment:
what about `universal`, wouldn't it be better?

Reply:
I think `default` is better since it is neutral from the semantics perspective

Reply:
In every other Devfile this is used to indicate language or framework, `default` looks weird to me.
Maybe my bigger problem with using `default` is that it could indicate that users use this Devfile **by default**, which I have many problems with. 

This devfile with UDI can still be helpful. But it should not be presented as a **default** Devfile.  I think that primarily we need to guide users towards using language or frameworks-specific Devfiles.  This universal devfile should be considered only if no other Devfile stack can be used or if the project uses multiple languages.

Reply:
> What if there are multiple languages and frameworks used in a particular devfile?
In general, I believe the projectType field should be optional, I do not understand why it is enforced at the moment on the registry end.

I agree with this. It is really weird that the registry requires something that is marked as optional in the Devfile spec. (cc: @elsony, @johnmcollier)


> this is about projectType which has nothing to do with devfile being the default per se, right?
universal is a weird name for projectType since it is very ambiguous

I was thinking about using `universal` as it is also in the image name.

The best would be to completely omit that, but we can do that :-(

We could keep `default` for now, fix the registry to not require it, and then remove it. But I know how this is going to play out; the change will never come and we will get stuck with `default` forever :-(

Reply:
+1 on changing it to `universal`. I think `default` is a bit deceiving as well since it implies being the best of all. We require that as part of the registry since we want consistency in the entries in the registry. There are other fields that are optional in the spec but required for the onboarding to the community registry for the same reasons, e.g. icons, description, etc.

Reply:
changed to `universal` even though I do not like the approach that registry enforces certain metadata that is odo specific